### DISCUSSION TITLE
Broad Design Concerns

### DISCUSSION THREAD

**Discussion by hgoldstein95** [clowdr-app/clowdr#188] [Ideas]
*I'm making this an issue so it's a bit easier for you to triage, but feel free to close it and move this somewhere else if needed.*

I have some broad, design-based feedback that I hope you'll keep in mind as Clowdr evolves. I imagine you've already thought about some of these points, but hopefully some of them are novel.

## Modelling Concepts for the User
At one point I heard an attendee say “there’s more on the screen now [versus at ICFP]”. This is patently false, but that was the perception. I propose that the issue here is concept modelling.

### Having fewer concepts isn’t necessarily better.
Unifying DMs, rooms, breakouts, etc into a single "concept" seems nice on the surface (there's only one thing to learn), but I suspect users don't see it that way. Users have already learned what a DM is---it’s a little chat box somewhere that sends a message to a particular person. By unifying DMs with other concepts, we actually put more burden on the user.

### Unified concepts in the back-end need not be unified in the front-end.
This is related to the previous point, but worth stating on its own. It’s totally fine that every chat room has an associated video room, but the “announcements” channel (for example) should not have that video room exposed. (I’ve seen confused people sitting around in the announcements room.)

## More Focus on High-Level Views
Users have high-level questions that they want to answer. Where is everyone? What are all of the events happening today? etc. The current interface has some of these views (if you know where to look), but they’re not up-front and obvious.

This could actually be fixed fairly easily with a few UI tweaks. I think the schedule should automatically start more “zoomed out” so the first thing the user sees is a birds-eye view. The rooms tab should sort social rooms to the top and set them apart, making it clear where to find people.

Personally I’d even like to see these things together somewhere. Dashboards make me feel in control of a piece of software.

## Minor Points
- *Viewer counts* - I don't think they actually add anything (especially when they're negative)
- *Smaller border radii* - Seems like a silly point, but I think shrinking the border radii a bit will make everything feel more professional
- *Fix elements relative to page size* - Getting a new message modifies everything on the page; chat windows should probably be fixed-size, rather than growing with the content

**Comment by rossng** [comment]
Thanks for this contribution! And sorry it's taken us a while to get around to sorting through our GitHub backlog.

Some thoughts from me:

> Having fewer concepts isn’t necessarily better.
> Unified concepts in the back-end need not be unified in the front-end.

Yes, I was wrong on this point. I believed that 'everything is a room' was a good idea back in November.

I think we have already improved this experience a _lot_. Chats appear in the right sidebar and you would never necessarily know that they were actually rooms under the surface. From a technical standpoint, I think we will continue unpicking the actual implementation of chat and rooms apart from one another.

> More Focus on High-Level Views

We have a bunch of work to do on the schedule view, in particular. I find that the new vertical layout helps a lot, but I think further UX/UI work will alleviate most of the current pain of using it.

The 'program' and 'social' rooms are now separated much more clearly, and we have plans to demarcate private rooms and public rooms more clearly.

I've also thought a little about how we can iterate on the conference landing page. At the moment it basically just displays an intro from the organisers and a list of content. I'm interested in some kind of 'discovery' mechanism that exposes what interesting stuff is happening now/today that you might not have noticed. What would you like to see?

> Viewer counts

I personally like having these on the livestreaming rooms. And they work _way_ better now! We also now have a live-upating list of 'who's here'.

However, I agree that they don't add much value to breakout rooms, especially when we already have the list of who's actually connected to the room as a video participant. We've talked about disabling the presence counts on these rooms.

> Fix elements relative to page size

I suspect this is fixed now.

**Reply by hgoldstein95** [reply]
Hi Ross! No worries about the delay, I didn't really expect a response at all, I just wanted to put my two cents in after POPL.

> I've also thought a little about how we can iterate on the conference landing page. At the moment it basically just displays an intro from the organisers and a list of content. I'm interested in some kind of 'discovery' mechanism that exposes what interesting stuff is happening now/today that you might not have noticed. What would you like to see?

As an organizer of the conference I was always trying to get a sense of what parts of the conference were active at a given time. Were people using the shuffle spaces? What rooms were in use? What talk was everyone watching? I think the discovery mechanism would be awesome too, but I'd be worried that something Clowdr didn't feel was important slipped by me. I'm looking for a no-frills way to see a lot of important things on one screen.

I know that's all super vague, but hopefully it's at least a little bit helpful.
